src/deadline_worker_agent/capabilities.py

```diff
@@ -103,9 +103,37 @@ def _get_gpu_count(*, verbose: bool = True) -> int:
         return int(output.decode().strip())
 
 
+def _parse_gpu_memory(lines: List, verbose: bool = True) -> List:
+
+    """
+    Parses the output of nvidia-smi to determine GPU memory.
+    Disabled GPUs will be assigned 0 MiB memory if they have no memory currently allocated for use.
+
+    Returns
+    -------
+    List
+        The memory available for each GPU.
+    """
+
+    mem_per_gpu: list[int] = []
+    for line in lines.splitlines():
+        try:
+            # Active GPU: nvidia-smi returns an integer followed by "MiB"
+            # Disabled GPU with no memory currently allocated: nvidia-smi returns "[N/A]"
+            mem_mib = int(line.replace("MiB", ""))
+        except ValueError:
+            if verbose:
+                _logger.warning(f"Could not detect GPU memory, non-numeric result returned by nvidia-smi: {line}")
+            mem_mib = 0
```

Comment:
```suggestion
            continue
```

How do you feel about simply skipping to the next line with `continue` if `nvidia-smi` returns a non integer response? This would allow us to keep the current pattern of not adding anything to the `mem_per_gpu` list when no GPU is detected.

Reply:
Looking closer, returning an empty list will cause an error further on, so I'm okay with keeping this as is.